py/torch_tensorrt/dynamo/conversion/_conversion.py

```diff
@@ -29,15 +29,21 @@ def infer_module_output_dtypes(
     truncate_double: bool = False,
 ) -> List[dtype]:
     """
-    inputs can be either arg_inputs or flattened input list. If it is flattened list, kwarg_inputs
-    should be None, as it is already included in the flattened input.
+    This function performs model inference to determine the output dtypes
+    and truncates them accordingly. inputs can be either arg_inputs or flattened input list.
+    If it is flattened list, kwarg_inputs should be None, as it is already included in the flattened input.
     """
+    # TODO: We can also determine output dtypes from the module.graph based on node metadata.
+    # However, our converter tests use fx.symbolic_trace which sometimes does not provide metadata,
```

Comment:
Can you explain this statement, would we rather use node metadata? For actual user use cases would this metadata be available. Since dynamo now assumes exported_program seems like the only other entry point is torch_compile where fx.graphs might show up. 

If we can use node metadata and fall back with warning to model inference that might be better. Also tests should move to using a pipeline more representative of the user pipeline graphs.